frictionless_ckan_mapper/ckan_to_frictionless.py

```diff
@@ -1,118 +1,123 @@
+'''
+Convert CKAN packages and resources to their
+Frictionless equivalent.
+'''
+# Standard library imports
 import json
 
+# Third-party library imports
 import slugify
 
+try:
+    json_parse_exception = json.decoder.JSONDecodeError
+except AttributeError:  # Testing against Python 2
+    json_parse_exception = ValueError
+
 
 class CKANToFrictionless:
+    '''Set of methods to convert CKAN packages and resources to their
+    Frictionless equivalent.'''
+
+    @staticmethod
+    def _resource_keys_to_remove(resource):
+        # TODO: Do we want to keep everything except `package_id` and
+        # `position`? The current test
+        # `test_resource_path_is_set_even_for_uploaded_resources`
+        # is not expecting `url_type` in its output.
+        resource_keys_to_remove = ['package_id', 'position', 'url_type']
+
+        # TODO: delete keys last as may be needed for something in processing
+        for k in resource_keys_to_remove:
+            if k in resource:
+                del resource[k]
+        return resource
 
-    resource_mapping = {
-        'size': 'bytes',
-        'mimetype': 'mediatype',
-        'url': 'path'
-    }
+    @staticmethod
+    def _remap_resource_ckan_to_frictionless(resource):
+        '''Remap differences from CKAN to Frictionless resource'''
+        resource_mapping = {
+            'size': 'bytes',
+            'mimetype': 'mediatype',
+            'url': 'path'
+        }
 
-    # TODO: Do we want to keep everything except `package_id` and
-    # `position`? The current test
-    # `test_resource_path_is_set_even_for_uploaded_resources`
-    # is not expecting `url_type` in its output.
-    resource_keys_to_remove = ['package_id', 'position', 'url_type']
+        for key, value in resource_mapping.items():
```

Comment:
What if the target resource already has that key? I've been having this problem integrating because I explicitly want to see what happens when a resource has both `url` and `path` set. If this is not something we can define a reasonable behavior for, we should probably raise an exception.